### DISCUSSION TITLE
Avoid huge changes in git history when regenerating parser

### DISCUSSION THREAD

**Discussion by edusporto** [tree-sitter/tree-sitter#3457] [Q&A] [answered]
Hi!

I'm writing a tree-sitter grammar for a language my team is developing. We host the grammar on GitHub, and currently the grammar's repository contains the `src` directory generated by the tree-sitter CLI. However, every time we change the grammar, the parser has to be regenerated, leading to a huge line difference between commits dealing with changes that might be very small.

Is there a way to deal with this problem? I'm looking mostly for what is the consensus regarding this issue, that is, how most people deal with it so our repository is similar to other existing tree-sitter grammars. It would be possible to just not commit the `src` folder, but I'm not sure if existing tools that user tree-sitter expect that folder to be there when installing grammars.

I wasn't able to find anything related to this in any other tree-sitter discussion forum, so I apologize if this has been asked before. I appreciate any help!

**Comment by WillLillis** [comment, marked as answer]
I'm not aware of any great workarounds to this currently, but there is a plan to address the issue eventually (see the "Mergeable Git Repos" item in #930).

**Reply by edusporto** [reply]
Thanks!

**Reply by sogaiu** [reply]
FWIW, a bit more detail and links to what some options are [here](https://github.com/sogaiu/ts-questions/blob/6e4a4b8342c6fab8cbbc14aa4e57fcd546229bca/questions/should-parser-source-be-committed/README.md) if interested.

**Reply by salimhisam505-collab** [reply]
اريد ان اصبح برمجي

**Reply by salimhisam505-collab** [reply]
اريد ان اصبح برمجي

**Comment by clason** [comment]
Compromise: omit the `parser.c` but commit the `grammar.json` -- this is much smaller, has semantic diffs, and can be used to generate a parser without having to have node or npm installed.
